module/users/helper.js

```diff
@@ -998,8 +999,117 @@ module.exports = class UserHelper {
       }
     });
   }
+
+  /**
+   * deleteUserPIIData function to delete users Data.
+   * @method
+   * @name deleteUserPIIData
+   * @param {userDeleteEvent} - userDeleteEvent message object 
+   * {
+        "entity": "user",
+        "eventType": "delete",
+        "entityId": 101,
+        "changes": {},
+        "created_by": 4,
+        "organization_id": 22,
+        "tenant_code": "shikshagraha",
+        "status": "INACTIVE",
+        "deleted": true,
+        "id": 101,
+        "username" : "user_shqwq1ssddw"
+    }* @returns {Promise} success Data.
+   */
+
+  static deleteUserPIIData(userDeleteEvent) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let userId = userDeleteEvent.id
+        // Throw an error if userId is missing
+				if (!userId) {
+					throw {
+						status: httpStatusCode.bad_request.status,
+						message: messageConstants.apiResponses.USER_ID_MISSING,
+					}
+				}
+
+        const filter = { createdBy: userId }
+
+        // Create update objects for each collection
+        const userProfileOps = _buildUpdateOperations(userProfileConfig, messageConstants.common.USER_PROFILE_KEY);
+        const nestedProfileOps = _buildUpdateOperations(userProfileConfig, messageConstants.common.USER_PROFILE_KEY_OBS);
+
+        // Merge updates for observation submissions
+        const observationUpdateOperations = {
+            ...(userProfileOps.$set || nestedProfileOps.$set ? { $set: { ...userProfileOps.$set, ...nestedProfileOps.$set } } : {}),
+            ...(userProfileOps.$unset || nestedProfileOps.$unset ? { $unset: { ...userProfileOps.$unset, ...nestedProfileOps.$unset } } : {})
+        }
+
+        //update in db
+        let updateDataStatus = await Promise.all([surveySubmissionsHelper.updateMany(filter,userProfileOps),
```

Comment:
@priyanka-TL check survey also. In case of projects we should check certificate data as well

Reply:
yes its already there